.gitlab/ci/.gitlab-ci.sdk-nightly.yml

```
@@ -264,20 +329,192 @@ demisto-sdk-nightly:run-end-to-end-tests:
     ARTIFACTS_FOLDER_INSTANCE: "${ARTIFACTS_FOLDER}/instance_${INSTANCE_ROLE}"
     ARTIFACTS_FOLDER_SERVER_TYPE: "${ARTIFACTS_FOLDER}/server_type_${SERVER_TYPE}"
     ENV_RESULTS_PATH: "${ARTIFACTS_FOLDER_SERVER_TYPE}/env_results.json"
+    FAIL_ON_ERROR: "true"
+    E2E_TEST_FOLDER: "xsoar"
   needs: ["demisto-sdk-nightly:xsoar-prepare-testing-bucket"]
   stage: run-instances
   script:
+    - !reference [.ssh-config-setup]
+
+    - !reference [.copy_env_results]
+
+    - section_start "Wait Until Server Ready" --collapsed
+    - |
+      [ -n "${NIGHTLY}" ] && IS_NIGHTLY=true || IS_NIGHTLY=false
+      python3 ./Tests/scripts/wait_until_server_ready.py  -n ${IS_NIGHTLY} --instance-role "${INSTANCE_ROLE}" || EXIT_CODE=$?
+    - section_end "Wait Until Server Ready"
+
+    - !reference [.download-demisto-conf]
+    - !reference [.secrets-fetch]
+
+    - section_start "End to End Tests"  --collapsed
     - |
       mkdir $ARTIFACTS_FOLDER/demisto-sdk
       git clone -b ${SDK_REF} --single-branch --depth 1 https://github.com/demisto/demisto-sdk.git $ARTIFACTS_FOLDER/demisto-sdk
 
-      TUNNEL_PORT=$(cat "${ENV_RESULTS_PATH}" | jq "map(select(.Role == \"$INSTANCE_ROLE\")) | .[].TunnelPort")
+      unset DEMISTO_API_KEY
       export DEMISTO_BASE_URL="https://$(cat "${ENV_RESULTS_PATH}" | jq -r '.[0].InstanceDNS')"
       echo "Server URL: $DEMISTO_BASE_URL"
-      unset DEMISTO_API_KEY # a temporary fix 
-      python3 -m pytest $ARTIFACTS_FOLDER/demisto-sdk/tests_end_to_end/ || EXIT_CODE=$?
+      python3 -m pytest ${ARTIFACTS_FOLDER}/demisto-sdk/tests_end_to_end/${E2E_TEST_FOLDER} || EXIT_CODE=$?
+    - section_end "End to End Tests"
+
+    - !reference [.prepare_for_test_content]
+
+    - section_start "Wait Until Server Ready"  --collapsed
+    - echo Going to sleep for 15 minutes to allow server finish indexing
+    - sleep-with-progress 900 30 "Sleeping... " 150
+    - echo "Done sleeping!"
+    - section_end "Wait Until Server Ready"
+
+    - !reference [.run_test_content]
+
+    - !reference [.assert_test_content]
+
+    - !reference [.cleanup_after_test_content]
+
     - job-done
     - exit $EXIT_CODE
+  after_script:
+    - !reference [.default-after-script]
+    - !reference [.install_ssh_keys]
+    - !reference [.ssh-config-setup]
+
+    - section_start "Destroy Instances" --collapsed
+    - python3 ./Tests/scripts/destroy_instances.py --artifacts-dir "${ARTIFACTS_FOLDER}" --env-file "${ENV_RESULTS_PATH}" --instance-role "${INSTANCE_ROLE}"
+    - destroy_instances_exit_code=$?
+    - |
+      if [ "${destroy_instances_exit_code}" -ne 0 ]; then
+        echo "Failed to destroy instances, exit code: ${destroy_instances_exit_code}"
+      fi
+    - section_end "Destroy Instances"
```

```suggestion
    - !reference [.destroy_xsoar_instances]
```
rebase with master and use this

---

you didn't accept my change. why?

---

Will push it with the rest of the cr comments